lib/screen/bloc/tzkt_transaction/tzkt_transaction_bloc.dart

```diff
@@ -26,10 +26,12 @@ class TZKTTransactionBloc
       newTokenItems ??= tokenItems;
       emit(TZKTTransactionState(newItems: [], isLastPage: true));
 
+      bool isLastPage = newOperationItems.length < event.pageSize;
       int? tokenNum;
-      if (newOperationItems.isNotEmpty) {
+      if (isLastPage) {
+        tokenNum = (newTokenItems?.length ?? 0) - 1;
+      } else {
         int lastOperationID = newOperationItems.last.getID();
```

Comment:
you don't check the `newOperationItems.isNotEmpty` anymore, so I think the `newOperationItems.last` might get error.